src/Splat.Microsoft.Extensions.DependencyInjection/MicrosoftDependencyResolver.cs

```diff
@@ -0,0 +1,213 @@
+﻿// Copyright (c) 2019 .NET Foundation and Contributors. All rights reserved.
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Splat.Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Microsoft DI implementation for <see cref="IDependencyResolver"/>.
+    /// </summary>
+    /// <seealso cref="IDependencyResolver" />
+    [SuppressMessage("Globalization", "CA1303", Justification = "Unnecessary warning")]
+    public class MicrosoftDependencyResolver : IDependencyResolver
+    {
+        private const string ImmutableExceptionMessage = "This container has already been built and cannot be modified.";
+        private readonly Lazy<IDictionary<string, IServiceScope>> _serviceScopes = new Lazy<IDictionary<string, IServiceScope>>(() => new Dictionary<string, IServiceScope>());
```

Comment:
Don't really see the advantage of using a lazy here since it's not getting created with values. Additionally it's not thread safe.